src/Scene.cpp

```diff
@@ -68,15 +63,22 @@ void Scene::CheckCollisions()
         player->ResetHit();
     }
 
-    // check if frog caught
-    if (duckyShape.intersects(frogShape))
+    // check collision with a frog
+    if (EntitiesIntersect(player->collisionComp, frog->frogShapeComp))
     {
         frogCounter->Increase();
         frog->Catch();
         frog->TeleportAwayFromPlayer(player->GetLocation());
     }
 }
 
+bool Scene::EntitiesIntersect(CircleCollisionComponent* component1, CircleCollisionComponent* component2)
```

Comment:
ok for now, but normally it should accept Entities as arguments, not components (even the name says EntitiesIntersect, not Components ;p). I also understand why it was easier to pull out the Collision Components into public interface.

Scene Graph will need to take it further, but for now it's a step in the correct direction

Reply:
Maybe if all entities like player, frog, greenEnemy had an universal type of entity with collisions from which CircleCollisionComponent could be taken and then the EntitiesIntersect() could have entity as an argument?
But it requires a bit of class rearranging and I'm not sure if it's worth the effort since it's going to be remade soon

However if such solution would even work and is timeworthy I can implement it :D